### PR TITLE
PLU-115: [SST-3] Refactor execute flow to not throw error

### PR DESCRIPTION
**Pull request by pregnantboy**
# TLDR;
This PR refactors the current execute flow mutation, both frontend and backend.

### Backend Changes:
1. Refactored `executeFlow` mutation to return `ExecutionStep` directly instead of `Step` and `dataOut`
2. `executeFlow` no longer returns the error by throwing. It now returns the last execution step ran together with the error details.
3. Modified `testRun` service to patch `testExecutionId` into the flow after processing all steps.

### Frontend Changes:
1. Updated `EXECUTE_FLOW` mutation to align with backend changes, refetching GET_TEST_EXECUTION_STEPS on complete.
2. Removed the `serializeErrors` function and using `errorDetails` directly.
3. Enhanced cache update logic in `TestSubstep` component to set the step status to 'completed' if the last execution step is successful, without refetching the entire flow.
4. Stores the last execution step error details in a state to display temporarily since the errorDetails could come from previous steps. (this is temporary since it would only test current step after all implementation is completed)

### To test
1. Click `Test Run`, the new variables should appear accordingly.
2. If the step was not tested, it should reflect tested after running test run.
3. If previous step failed, it should show the previous step error temporarily.
4. If current step failed, it should show the current step error even after refreshing.

### CODE REVIEW

**Comment by pregnantboy** [comment]
* **#724** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/724?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#708** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/708?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#707** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/707?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#706** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/706?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#680** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/680?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#679** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/679?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#678** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/678?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pregnantboy and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/src/services/test-run.ts [hunk 1]
```diff
@@ -86,6 +86,10 @@ const testRun = async (options: TestRunOptions) => {
       nextStepId = nextStep?.id
     }
   }
+
+  await flow.$query().patch({
```
Comment (m0nggh) [inline packages/backend/src/services/test-run.ts]: wait sorz is it me or this will never occur? should this be done at the start instead

**Review by m0nggh — CHANGES_REQUESTED** [review]
need to clarify one comment, will need to re-test again with the `test_execution_id`.

but tested without the `test_execution_id` column being filled and everything works as intended:
- `we couldn't find any test data` appears by default for untested steps
- clicking test run will update with the new variables
- refreshing the page will show the new updated executionstep
- error shows if test step fails
- previous error appears for current step regardless of whether it works

**Review by m0nggh — APPROVED** [review]
ok tested that it works now, lgtm

**Comment by pregnantboy** [comment]
### Merge activity

* **Aug 27, 6:30 AM EDT**: @pregnantboy started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/680).
* **Aug 27, 6:36 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/680) couldn't merge this PR because it had conflicts with the trunk branch.
